mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergInputFormat.java

```diff
@@ -59,12 +57,12 @@ public class HiveIcebergInputFormat extends MapredIcebergInputFormat<Record>
   private static final Logger LOG = LoggerFactory.getLogger(HiveIcebergInputFormat.class);
   private static final String HIVE_VECTORIZED_RECORDREADER_CLASS =
       "org.apache.iceberg.mr.hive.vector.HiveIcebergVectorizedRecordReader";
-  private static final DynConstructors.Ctor<AbstractMapredIcebergRecordReader>
-      HIVE_VECTORIZED_RECORDREADER_CTOR;
+  private static DynConstructors.Ctor<AbstractMapredIcebergRecordReader>
+      hiveVectorizedRecordreaderCtor;
```

Comment:
not sure why this is non-final now?

Reply:
`HiveIcebergVectorizedRecordReader` is in `iceberg-hive3` module and can not be loaded in `iceberg-mr` tests.  Hence, we need to catch the exception and reassign it to `null`. Same for `HiveVectorizedReader`.